Assets/_Laz/Scripts/ScriptableObject/Laz/LazMovementPropertyScriptableObject.cs

```diff
@@ -17,30 +17,40 @@ public interface ILazMovementProperty
     [InlineEditor]
     public class LazMovementPropertyScriptableObject : ScriptableObject, ILazMovementProperty
     {
-        [SerializeField]
-        private float _acceleration = 1f;
-        [SerializeField]
-        private float _deceleration = 1f;
-        [SerializeField]
-        private float _baseMaxSpeed = 5f;
-        [SerializeField]
-        [Range(0.005f, 0.1f)]
-        private float _curvatureRate = 0.1f;
-        [SerializeField]
-        private float _lazMaxSpeed = 10f;
+        // [SerializeField]
+        // private float _acceleration = 1f;
+        // [SerializeField]
+        // private float _deceleration = 1f;
+        // [SerializeField]
+        // private float _baseMaxSpeed = 5f;
+        // [SerializeField]
+        // [Range(0.005f, 0.1f)]
+        // private float _curvatureRate = 0.1f;
+        // [SerializeField]
+        // private float _lazMaxSpeed = 10f;
+        //
+        // [Header("Boost Properties")] 
+        // [SerializeField]
+        // private float _boostTimeLimit = 2f;
+        // [SerializeField]
+        // private float _boostSpeed = 20f;
+
+        public float acceleration = 1f;
+        public float deceleration = 1f;
+        public float baseMaxSpeed = 5f;
+        [Range(0.005f, 0.1f)] public float curvatureRate = 0.1f;
+        public float lazMaxSpeed = 10f;
 
         [Header("Boost Properties")] 
-        [SerializeField]
-        private float _boostTimeLimit = 2f;
-        [SerializeField]
-        private float _boostSpeed = 20f;
+        public float boostTimeLimit = 2f;
+        public float boostSpeed = 20f;
```

Comment:
I, unfortunately, had to make all fields in the ScriptableObject public, cause JsonUtility can only serialize and Deserialize Public data. But because of the interface being used in a lot of different scripts as well as the Unit Tests, I just decided to keep it. Suggestions into if we should alter how this works?

Reply:
So this is generic engineering "issue". 

most of the time if you're client is interacting with a third party library - DB, JSON, Endpoint. You want to have a clear separation between you're code and third party code.  So you abstract your/our code one more layer into something called a DTO - Data Transfer Object. 

So you have this LazMovementPropertyScriptableObject, So you can create a separate LazMovementDTO, with all the fields inside public, and you pass this DTO object to the JSON writer to do whatever. 

And then when you deserialize it, you can then use the deserialized DTO to create a new separate LazMovementPropertyScriptableObject

More Details, or you can Google:
https://stackabuse.com/data-transfer-object-pattern-in-java-implementation-and-mapping/
The Data Transfer Object Design Pattern is one of the enterprise application architecture patterns that calls for the use of objects that aggregate and encapsulate data for transfer